src/opm/parser/eclipse/EclipseState/NumericalAquifer.cpp

```diff
@@ -0,0 +1,327 @@
+    for (const auto& pair : this->aquifers_) {
+        auto trans_aquifer = pair.second.transToRemove(grid);
+        for (size_t i = 0; i < 3; ++i) {
+            trans[i].merge(trans_aquifer[i]);
+        }
+    }
+    return  trans;
+}
+
+void NumericalAquifers::
+appendNNC(const EclipseGrid &grid, const FieldPropsManager &fp, NNC &nnc) const {
+    for (const auto& pair : this->aquifers_) {
+        pair.second.appendNNC(grid, fp, nnc);
+    }
+}
+
+const std::unordered_map<size_t, const NumericalAquiferCell>& NumericalAquifers::
+aquiferCells() const {
+    return this->aquifer_cells_;
+}
+
+bool NumericalAquifers::hasCell(const size_t cell_global_index) const {
+    const auto& cells = this->aquifer_cells_;
+    return (cells.find(cell_global_index) != cells.end());
+}
+
+const NumericalAquiferCell& NumericalAquifers::getCell(const size_t cell_global_index) const {
+    assert(this->hasCell(cell_global_index));
+    return this->aquifer_cells_.at(cell_global_index);
+}
+
+using AQUNUM = ParserKeywords::AQUNUM;
+NumericalAquiferCell::NumericalAquiferCell(const DeckRecord& record, const EclipseGrid& grid, const FieldPropsManager& field_props)
+   : aquifer_id( record.getItem<AQUNUM::AQUIFER_ID>().get<int>(0) )
+   , I ( record.getItem<AQUNUM::I>().get<int>(0) - 1 )
+   , J ( record.getItem<AQUNUM::J>().get<int>(0) - 1 )
+   , K ( record.getItem<AQUNUM::K>().get<int>(0) - 1 )
+   , area (record.getItem<AQUNUM::CROSS_SECTION>().getSIDouble(0) )
+   , length ( record.getItem<AQUNUM::LENGTH>().getSIDouble(0) )
+   , permeability( record.getItem<AQUNUM::PERM>().getSIDouble(0) )
+{
+    const auto& cell_depth = field_props.cellDepth();
+    const auto& poro = field_props.get_double("PORO");
+    const auto& pvtnum = field_props.get_int("PVTNUM");
+    const auto& satnum = field_props.get_int("SATNUM");
+
+    this->global_index = grid.getGlobalIndex(I, J, K);
+
+    if ( !record.getItem<AQUNUM::PORO>().defaultApplied(0) ) {
+        this->porosity = record.getItem<AQUNUM::PORO>().getSIDouble(0);
+    } else {
+        this->porosity = poro[global_index];
+    }
+
+    if ( !record.getItem<AQUNUM::DEPTH>().defaultApplied(0) ) {
+        this->depth = record.getItem<AQUNUM::DEPTH>().getSIDouble(0);
+    } else {
+        this->depth = cell_depth[global_index];
+    }
+
+    if ( !record.getItem<AQUNUM::INITIAL_PRESSURE>().defaultApplied(0) ) {
+        this->init_pressure = record.getItem<AQUNUM::INITIAL_PRESSURE>().getSIDouble(0);
+    }
+
+    if ( !record.getItem<AQUNUM::PVT_TABLE_NUM>().defaultApplied(0) ) {
+        this->pvttable = record.getItem<AQUNUM::PVT_TABLE_NUM>().get<int>(0);
+    } else {
+        this->pvttable = pvtnum[global_index];
+    }
+
+    if ( !record.getItem<AQUNUM::SAT_TABLE_NUM>().defaultApplied(0) ) {
+        this->sattable = record.getItem<AQUNUM::SAT_TABLE_NUM>().get<int>(0);
+    } else {
+        this->sattable = satnum[global_index];
+    }
+
+    this->pore_volume = this->length * this->area * this->porosity;
```

Comment:
I guess this is on purpose - but you can ask the grid to get a proper cell volume(s)?

Reply:
No. It is this one to overwrite the grid one.